components/new-contribution-flow/ContributionFlowMainContainer.js

```diff
@@ -54,6 +57,57 @@ class NewContributionFlowMainContainer extends React.Component {
     }
   };
 
+  getLoggedInUserDefaultContibuteProfile() {
+    if (this.props.contributeAs) {
+      const otherProfiles = this.getOtherProfiles();
+      const contributorProfile = otherProfiles.find(profile => profile.slug === this.props.contributeAs);
+      if (contributorProfile) {
+        return contributorProfile;
+      }
+    }
+    if (get(this.props.mainState, 'stepProfile')) {
+      return this.props.mainState.stepProfile;
+    }
+    if (this.props.LoggedInUser) {
+      return this.getPersonalProfile();
+    }
+  }
+
+  /** Returns logged-in user profile */
+  getPersonalProfile() {
+    const { LoggedInUser } = this.props;
+    if (!LoggedInUser) {
+      return {};
+    }
+
+    return { email: LoggedInUser.email, image: LoggedInUser.image, ...LoggedInUser.collective };
+  }
+
+  /** Return an array of any other associated profile the user might control */
+  getOtherProfiles() {
+    const { LoggedInUser, collective } = this.props;
+    if (!LoggedInUser) {
+      return [];
+    }
+
+    return LoggedInUser.memberOf
```

Comment:
We don't have to do it on this PR, but since we're moving the query to V2 it would be great to load the contribution profiles dynamically (in `pages/new-contribution-flow.js`). https://github.com/opencollective/opencollective/issues/2427 is actually listed in https://github.com/opencollective/opencollective/projects/39 TODOs.

Reply:
How do you mean? You want to do a single, separate query that just fetches Collectives/Orgs that the logged in user is a member of that executes only when we get to `StepProfile` and the person is logged in?

This makes sense but won't the `memberOf` block still be part of the `LoggedInUser` query provided by `withUser` anyways? Do you mean we should take that out of `loggedInUserQuery` in `lib/graphql/queries.js` and refactor any other files using it too?

Just trying to understand 😊

Reply:
> Do you mean we should take that out of loggedInUserQuery in lib/graphql/queries.js and refactor any other files using it too?

It's the long-term goal. Relying on the `LoggedInUser` provided by `withUser` for the data is not great, because:
- The data will often be outdated
- We're fetching too much info when logging in (slowing down the query)

For now, a good first step is to avoid relying on this data for new developments. We could either fetch the data from `StepDetails`, or from `pages/new-contribution-flow.js` (in the page query).

But this is an enhancement for later, we don't have to do it today and it won't block the release.